snmp/README.md

```diff
@@ -19,12 +19,66 @@ The SNMP check is included in the [Datadog Agent][1] package. No additional inst
 
 ### Configuration
 
-The Datadog SNMP check auto-discovers network devices on a provided subnet, and collects metrics using Datadog's sysOID mapped device profiles.
+Edit the configuration options in the `snmp.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][2]. See the [sample snmp.d/conf.yaml][3] for all available configuration options.
 
-Edit the subnet and SNMP version in the `snmp.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][2]. See the [sample snmp.d/conf.yaml][3] for all available configuration options.
+The Datadog SNMP check supports collecting metrics from individual devices, or auto-discovering devices on entire subnets.
+
+Which collection strategy to choose mainly depends on how many devices are present on your network, and how dynamic the network is (i.e. how frequently devices are added or removed):
+
+- For small and mostly static networks, see [Monitoring individual devices](#monitoring-individual-devices).
+- For larger and/or dynamic networks, see [Autodiscovery](#autodiscovery).
+
+Regardless of the collection strategy, you can leverage Datadog's [`sysObjectID` mapped device profiles](#sysobjectid-mapped-device-profiles) to automatically collect relevant metrics from your devices.
+
+#### Monitoring individual devices
+
+The easiest way to get started with the SNMP integration is to point it at the IP address of an SNMP device.
+
+For a minimal setup, we'll use the `generic-router` profile. It collects generic metrics exposed by most network devices (such as ones in `IF-MIB`, `IP-MIB`, etc).
+
+For SNMPv2 devices:
+
+1. Configure an instance specifying i) the IP address of the device, ii) the `community` string of the device, and iii) the `generic-router` profile:
+
+    ```yaml
+    instances:
+      - ip_address: "<IP_ADDRESS>"
+        community: "<COMMUNITY>"
+        profile: generic-router
```

Comment:
Out of curiosity, what's the purpose of guiding customers toward defining profiles inline to the config? Don't we automatically match to the most specific profile without specifying anything?

Reply:
You're right, without specifying a `profile` then the check will fetch the `sysObjectID` and match a profile, and worst case scenario no specific device profile matches and the check will use `generic-router`.

That said, I think this README needs some tutorial steps, where we take users by the hand to see their first metrics in Datadog. The goal is to improve onboarding UX, and reduce support burden.

We can definitely document profile auto-matching later (I think we need a dedicated section about this, because it's applicable both for autodiscovery and single device monitoring).

But for these simple first steps I wanted to keep things simple, so we can 100% guarantee that if a user set things up like this, restart their Agent, they'll start seeing metrics in Datadog right away, and they'll be happy that they were able to do it on their own.

Does that make sense?

Reply:
> You're right, without specifying a profile then the check will fetch the sysObjectID and match a profile, and worst case scenario no specific device profile matches and the check will use generic-router.

I'm actually _not sure_ about this… Need to check.

Reply:
Thinking through it once again, we can't actually guarantee 100% that `generic-router` will result in metrics being collected… Eg IIRC Proliant (or iDRAC?) devices don't expose IF-MIB metrics (but instead a proprietary variant).

Working on a rewording, then. :-)

Reply:
Thanks for these changes -- looks good to me. 

At the BARE minimum I think everything should at least get like, TCP metrics right? I guess we won't be perfect in all cases, but I do think that devices that really don't collect metrics from any of our profiles will be a big exception to the rule

Reply:
I haven't found a common MIB that all devices expose (eg TCP metrics aren't exposed by 5 BIG IP, NetApp, Meraki, etc), so the way I worded it now is hinting readers to look at the metrics in the matched profile. :-)